### DISCUSSION TITLE
How to set watch ignore *.extension?

### DISCUSSION THREAD

**Discussion by cekvenich2** [11ty/eleventy#2032] [Uncategorized]
Hi all. I have build and source same, and when it builds(pug) the html files trigger a build - an endless loop.

How do I set watcher to ignore .html files?
This does not work:
eleventyConfig.ignores.add("*.html");

tia. Vic

**Comment by pdehaan** [comment]
I believe that dynamic ignores pattern only works in v1 (currently in beta).

I think there might be a few ways to solve this:
- https://www.11ty.dev/docs/config/#template-formats
- Explicitly set an input and output folder in your .eleventy.js config file, and 11ty won’t try processing your generated output files. 
- possibly tweak the .eleventyignore file to ignore *.html globs

I’d probably recommend option 1 or 2.

**Comment by cekvenich2** [comment]
I am using v1, how would I do it in v1?

(I tried the other ways and no joy)

**Reply by pdehaan** [reply]
Interesting. I tried w/ eleventy v1.0.0-beta.2 and setting the input/output folders worked for me.

```js
.eleventy.js
module.exports = (eleventyConfig) => {
  return {
    dir: {
      input: "src",
      output: "www",
    }
  };
};
```

```pug
---
# src/index.pug
title: I forget how Pug works
---

p.foo=title

footer
  p=eleventyVersion
``` 

```js
// src/_data/eleventyVersion.js
module.exports = require("@11ty/eleventy/package.json").version;
```

### OUTPUT

```sh
tree -aI node_modules

.
├── .eleventy.js
├── package-lock.json
├── package.json
├── src
│   ├── _data
│   │   └── eleventyVersion.js
│   └── index.pug
└── www
    └── index.html
```

And since the Pug templates are written to the output folder (./www) Eleventy ignores them.

Do you have a public repo somewhere I can clone and take a look at? It might be something w/ your config if the pug files are being written to the input folder where they are being reprocessed.

**Reply by pdehaan** [reply]
It sounds like you're hitting https://www.11ty.dev/docs/languages/html/#same-input-output
But hard to tell without seeing the config file.

**Comment by pdehaan** [comment]
Eleventy should ignore the output folder by default, but if you are using the same input/output folder, this seems to work for me locally (although not my preferred solution):

```sh
npx eleventy --input=. --output=. --formats=pug
```

Or, if you want to specify it in your .eleventy.js config file (per https://www.11ty.dev/docs/config/#template-formats):

```js
module.exports = (eleventyConfig) => {
  eleventyConfig.setTemplateFormats("pug,md");

  return {
    dir: {
      input: ".",
      output: ".",
      
    },
    // templateFormats: ["pug", "md"]
  };
};
```

**Comment by cekvenich2** [comment]
Thx, The same output/input folder I already set (as ln -s so it copies all files, else it misses some otherwise).

https://gitlab.com/cekvenich2/ubay

That is the settings there.

The issue is that after it builds pug: it builds again html in an endless loop when I watch. Build works fine.

**Comment by cekvenich2** [comment]
I just used v1 w/ passthroughall

(asiude I wish instead of while list I could just black list files, like
pass all except *.pug ).

**Comment by pdehaan** [comment]
Odd, I cloned and installed Eleventy v1 beta and it built fine for me on
macOS Big Sur. Wonder if it’s related to the symlinked folder

**Comment by cekvenich2** [comment]
thanks for the help!!!
